src/store/root.store.ts

```diff
@@ -127,10 +139,10 @@ export default class RootStore {
 
   @action
   infiniteFetchLimit(count: number): void {
-    this.newsFlashFetchLimit += count;
-    const { newsFlashCollection, newsFlashFetchLimit } = this;
-    if (newsFlashCollection.length < newsFlashFetchLimit - count) return;
-    this.filterNewsFlashCollection();
+    this.newsFlashFetchOffSet += count;
+    const { newsFlashCollection, newsFlashFetchOffSet, newsFlashActiveFilter } = this;
+    if (newsFlashCollection.length < newsFlashFetchOffSet - count) return;
+    this.filterNewsFlashCollection(newsFlashFetchOffSet, newsFlashActiveFilter);
```

Comment:
1. I see no point in this restructure, 
2. the condition + return is confusing 
3. no need to provide filterNewsFlashCollection with parameters of the same object ("this")
how about this?
```
infiniteFetchLimit(count: number) {
    this.newsFlashFetchOffSet += count;
    const { newsFlashCollection, newsFlashFetchOffSet, newsFlashActiveFilter } = this;
    if (this.newsFlashCollection.length >= this.newsFlashFetchOffSet - count) {
        this.filterNewsFlashCollection();
    }
}
```

Reply:
Regarding point 3, I think it's more efficient and readable data flow, provide param to function then just put global object directly, especially that one param come in any case from outside
@action setActiveNewsFlashFilter(filter: SourceFilterEnum) {
    if (filter !== this.newsFlashActiveFilter) {
      this.newsFlashActiveFilter = filter;
      this.newsFlashCollection = [];
      this.newsFlashFetchOffSet = 0;
      this.filterNewsFlashCollection(this.newsFlashFetchOffSet, filter);
    }
  }